collect_app/src/test/java/org/odk/collect/android/http/mock/MockHttpClientConnection.java

```diff
@@ -0,0 +1,68 @@
+package org.odk.collect.android.http.mock;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.odk.collect.android.http.HttpCredentialsInterface;
+import org.odk.collect.android.http.HttpGetResult;
+import org.odk.collect.android.http.HttpHeadResult;
+import org.odk.collect.android.http.OpenRosaHttpInterface;
+import org.odk.collect.android.utilities.ResponseMessageParser;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MockHttpClientConnection implements OpenRosaHttpInterface {
+
+    private boolean getHttpShouldReturnNull;
+
+    @NonNull
+    @Override
+    public HttpGetResult get(@NonNull URI uri, @Nullable String contentType, @Nullable HttpCredentialsInterface credentials) throws Exception {
+        if (getHttpShouldReturnNull) {
```

Comment:
I am trying to think of a way where using this wouldn't be required but we could still have the desired result. Such as using `Mockito` to return `null` when called. Just an idea, might be an overkill but I just like using a much test related apis as possible to make the intentions clearer.

Reply:
So i've modified the code to mock the get() method when the testGetXMLDocumentErrorResponse() runs. I think it's cleaned up the code quite nicely.